fpmsyncd/routesync.cpp

```diff
@@ -581,7 +581,8 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
     char master_name[IFNAMSIZ] = {0};
 
     /* if the table_id is not set in the route obj then route is for default vrf. */
-    if (master_index)
+    if (master_index &&
+        (master_index != RT_TABLE_MAIN) && (master_index != RT_TABLE_DEFAULT))
```

Comment:
what is trigger of this change ?

Reply:
Main difference between default fpmsyncd and "fpmsyncd -l net" behavior is related to VRF support. SONiC modifies actual netlink message from FRR to send VRF ifindex value in RTA_RT_TABLE_ID attribute, while Linux kernel netlink message is unmodified and contains the route table ID in RTA_RT_TABLE_ID attribute. This prevents "-l net" option being used in scenarios that test routes with VRF.

Reply:
@abdosi I can change this logic to just check a boolean to see if Linux Netlink connection is used and skip VRF if it is.